src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
+      animal.health -= 50;
+
+      Animal.alive = Animal.alive.filter(checkAnimal => checkAnimal.health > 0);
```

Comment:
Let's move the `delete` method into a instance of the `Animal` class
And add a check that if the animal you bite has health 0 or less, then we delete it

```
if (animal.health <= 0) {
  Animal.removeFromAlive(animal);
}
```